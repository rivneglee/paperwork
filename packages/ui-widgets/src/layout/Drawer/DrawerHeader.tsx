import React, { FunctionComponent, SyntheticEvent } from 'react';
import { IconButton } from '../../form/IconButton';
import Icons from '../../graphic/Icons';

interface Props {
  onClickCloseButton?: (e: SyntheticEvent) => void;
  placement: 'left' | 'right';
}

const DrawerHeader: FunctionComponent<Props> = ({
  children,
  onClickCloseButton,
  placement,
}) => (
  <>
    <div className="pw-drawer-header">
      {children}
    </div>
    <IconButton onClick={onClickCloseButton}>
      {
        placement === 'left' && <Icons.ArrowLeft />
      }
      {
        placement === 'right' && <Icons.ArrowRight />
      }
    </IconButton>
  </>
);

export default DrawerHeader;
