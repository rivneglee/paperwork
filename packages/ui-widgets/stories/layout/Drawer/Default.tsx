import React, { useState } from 'react';

import { Drawer, Icons, List } from '../../../src';

export default () => {
  const [isShow, setIsShow] = useState(false);
  const close = () => setIsShow(false);
  const open = () => setIsShow(true);
  return (
    <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
      <div onClick={open}>
        <Icons.Menu />
      </div>
      <Drawer header={
        <h3>Paperwork</h3>
      } isShow={isShow} onClose={close}>
        <List>
          <List.Item onClick={close} icon={<Icons.Form />}>
            Forms
          </List.Item>
          <List.Item onClick={close} icon={<Icons.Chart />}>
            Reports
          </List.Item>
        </List>
      </Drawer>
    </div>
  );
};
