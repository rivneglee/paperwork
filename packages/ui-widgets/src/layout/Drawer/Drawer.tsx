import React, { ReactElement, SyntheticEvent, FunctionComponent } from 'react';
import classNames from 'classnames';

import { Card } from '../Card';
import DrawerHeader from './DrawerHeader';

interface Props {
  isShow: boolean;
  header?: ReactElement<any>;
  footer?: ReactElement<any>;
  onClose?: (e: SyntheticEvent) => void;
  placement?: 'left' | 'right';
  useOverlay?: boolean;
}

const Drawer: FunctionComponent<Props> = ({
  isShow,
  onClose,
  header,
  footer,
  children,
  placement = 'left',
  useOverlay = true,
}) => (
  <>
    <Card
      className={
        classNames(`pw-drawer pw-drawer--${placement}`, { ['pw-drawer--opened']: isShow })
      }
      header={
        <DrawerHeader placement={placement} onClickCloseButton={onClose}>
          {header}
        </DrawerHeader>
      }
      footer={
        footer
      }
    >
      { children }
    </Card>
    {
      useOverlay && (
        <div
          className={classNames('pw-drawer-overlay', { ['pw-drawer-overlay--show']: isShow })}
          onClick={onClose}
        >
        </div>
      )
    }
  </>
);

export default Drawer;
