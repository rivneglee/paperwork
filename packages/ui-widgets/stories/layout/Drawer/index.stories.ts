import { storiesOf } from '@storybook/react';

import Default from './Default';
import WithPlacement from './WithPlacement';

storiesOf('Drawer', module)
  .add('default', Default)
  .add('withPlacement', WithPlacement);
