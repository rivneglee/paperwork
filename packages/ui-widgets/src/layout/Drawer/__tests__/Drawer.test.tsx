import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Drawer from '../Drawer';

describe('IconButton', () => {
  const Content = () => <>Content</>;

  it('should render drawer when it opened', () => {
    const wrapper: ShallowWrapper = shallow(
      <Drawer isShow={true}>
        <Content />
      </Drawer>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not render drawer when it closed', () => {
    const wrapper: ShallowWrapper = shallow(
      <Drawer isShow={false}>
        <Content />
      </Drawer>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with right style when place to right', () => {
    const wrapper: ShallowWrapper = shallow(
      <Drawer isShow={false} placement="right">
        <Content />
      </Drawer>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
