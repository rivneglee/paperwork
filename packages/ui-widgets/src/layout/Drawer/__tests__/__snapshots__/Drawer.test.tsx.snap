// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should not render drawer when it closed 1`] = `
<Fragment>
  <Card
    className="pw-drawer pw-drawer--left"
    header={
      <DrawerHeader
        onClickCloseButton={undefined}
        placement="left"
      />
    }
  >
    <Content />
  </Card>
  <div
    className="pw-drawer-overlay"
  />
</Fragment>
`;

exports[`IconButton should render drawer when it opened 1`] = `
<Fragment>
  <Card
    className="pw-drawer pw-drawer--left pw-drawer--opened"
    header={
      <DrawerHeader
        onClickCloseButton={undefined}
        placement="left"
      />
    }
  >
    <Content />
  </Card>
  <div
    className="pw-drawer-overlay pw-drawer-overlay--show"
  />
</Fragment>
`;

exports[`IconButton should render with right style when place to right 1`] = `
<Fragment>
  <Card
    className="pw-drawer pw-drawer--right"
    header={
      <DrawerHeader
        onClickCloseButton={undefined}
        placement="right"
      />
    }
  >
    <Content />
  </Card>
  <div
    className="pw-drawer-overlay"
  />
</Fragment>
`;
